setup.py

```diff
@@ -173,9 +173,33 @@ class MyInstallData (install_data, object):
 
     def run (self):
         """Adjust permissions on POSIX systems."""
+        self.install_translations()
         super(MyInstallData, self).run()
         self.fix_permissions()
 
+    def install_translations (self):
+        """Install compiled gettext catalogs."""
+        # A hack to fix https://github.com/linkcheck/linkchecker/issues/102
+        i18n_files = [
+            (dir, files) for (dir, files) in self.data_files
+            if 'LC_MESSAGES' in dir
+        ]
+        self.data_files = [
+            (dir, files) for (dir, files) in self.data_files
+            if 'LC_MESSAGES' not in dir
+        ]
+        for dir, files in i18n_files:
+            dir = util.convert_path(dir)
+            if not os.path.isabs(dir):
+                dir = os.path.join(self.install_dir, dir)
+            elif self.root:
+                dir = util.change_root(self.root, dir)
+            self.mkpath(os.path.dirname(dir))  # <-- the magic happens here
```

Comment:
From https://github.com/linkcheck/linkchecker/commit/d94bd0ee0660a989eb49d4179ed1db7af9c766d9#commitcomment-25354514:
> expand. what magic? what does this do? what problem does it solve?

This is essentially copied and pasted from [distutils.command.install_data](https://github.com/python/cpython/blob/b694770a2b23cd485c98bf673a8b2dc1a865d9df/Lib/distutils/command/install_data.py#L45-L75), except I pruned some branches (`if isinstance(f, str)`, since f is always a tuple for us, and `if f[1] == []`, since we're not creating empty directories here), changed the outer loop variables to destructure tuples, and changed one line of actual code.  The "magic" comment pinpoints that line: distutils's install_data.run() does a self.mkpath(dir) at this point, and we do `self.mkdir(os.path.dirname(dir))`.  Here's why:

For .mo files `dir` is something like `/usr/local/share/locale/fr/LC_MESSAGES/linkchecker.mo`, and `files` is `['po/fr.mo']`.  We want to `mkdir /usr/local/share/locale/fr/LC_MESSAGES` and copy `po/fr.mo` into `/usr/local/share/locale/fr/LC_MESSAGES/linkchecker.mo`.  `self.copy_file(src, dst)` behaves like shell's `cp`: if dst is a directory, it creates a file with the same name under it, and if it's not, it's taken as the destination filename.  So 

    self.copy_file('po/fr.mo', '/usr/local/share/locale/fr/LC_MESSAGES/linkchecker.mo')

will do what we want *as long as `/usr/local/share/locale/fr/LC_MESSAGES/linkchecker.mo`* is not a directory.

Reply:
don't tell me, say so in the code. :)

that said:

> The "magic" comment pinpoints that line: distutils's install_data.run() does a self.mkpath(dir) at this point, and we do self.mkdir(os.path.dirname(dir)).

don't we do `self.mkpath`?

and why don't we just call `copy_file()` ourselves? if we're going to coerce `install_data()` to try and fit a square peg in a round hole, why not just add a new `install_trans` command that would simply install translations in the expected location? because at this point it seems like we're reinventing `install_data` in everything but the name...

this, in turn, could eventually be shared with other projects more cleanly...

Reply:
oh wait we do call copy_file, nevermind. :)